### PR TITLE
Update adapter to work with EC3 API Update

### LINKED ISSUE #107: Update API calls to EC3's new API
#### Description:

Update all API calls and deprecate the current user auth method to prompt the user to provide an API key instead of user credentials.

### PR DESCRIPTION
**Pull request by michaelhoehn**
### Issues addressed by this PR

Closes #107 

### Test files

[here](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/CarbonQueryDatabase_Toolkit/%23108%20Update%20API%20calls.gh?csf=1&web=1&e=rjFurC)

### CODE REVIEW

File: CarbonQueryDatabase_Engine/Compute/APIToken.cs [hunk 1]
```diff
@@ -34,15 +34,16 @@ namespace BH.Engine.Adapters.CarbonQueryDatabase
 {
     public static partial class Compute
     {
-        [Description("Returns a bearer token for the CarbonQueryDatabase system from the provided username and password")]
+        [PreviousVersion("6.2", "BH.Engine.Adapters.CarbonQueryDatabase.CQDBearerToken(string username = \"\", string password = \"\")")]
```
Comment (FraserGreenroyd) [inline CarbonQueryDatabase_Engine/Compute/APIToken.cs:37]:
```suggestion
        [PreviousVersion("6.2", "BH.Engine.Adapters.CarbonQueryDatabase.CQDBearerToken(System.String, System.String)")]
```

File: CarbonQueryDatabase_Engine/Create/GetRequest.cs [hunk 2]
```diff
@@ -34,12 +34,14 @@ public static partial class Create
         /***************************************************/
         /**** Public  Method                            ****/
         /***************************************************/
+
+        [PreviousVersion("6.2", "BH.Engine.Adapters.CarbonQueryDatabase.CarbonQuery(string username = \"\", string password = \"\"), BH.oM.Adapters.CarbonQueryDatabase.CQDConfig parameters = null")]
```
Comment (FraserGreenroyd) [inline CarbonQueryDatabase_Engine/Create/GetRequest.cs:38]:
```suggestion
```

The signature hasn't changed so no previous version attribute needed.

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

File: CarbonQueryDatabase_Adapter/CarbonQueryDatabase_Adapter.cs [hunk 3]
```diff
@@ -41,23 +41,23 @@ public partial class CarbonQueryDatabaseAdapter : BHoMAdapter
         /**** Constructors                              ****/
         /***************************************************/
 
+        [PreviousVersion("6.2", "BH.Adapter.CarbonQueryDatabase(System.String, System.String, System.Bool)")]
```
Comment (FraserGreenroyd) [inline CarbonQueryDatabase_Adapter/CarbonQueryDatabase_Adapter.cs:44]:
```suggestion
        [PreviousVersion("6.2", "BH.Adapter.CarbonQueryDatabaseAdapter(System.String, System.String, System.Bool)")]
```

File: CarbonQueryDatabase_Engine/Compute/APIToken.cs [hunk 4]
```diff
@@ -34,15 +34,16 @@ namespace BH.Engine.Adapters.CarbonQueryDatabase
 {
     public static partial class Compute
     {
-        [Description("Returns a bearer token for the CarbonQueryDatabase system from the provided username and password")]
+        [PreviousVersion("6.2", "BH.Engine.Adapters.CarbonQueryDatabase.CQDBearerToken(System.String, System.String)")]
```
Comment (FraserGreenroyd) [inline CarbonQueryDatabase_Engine/Compute/APIToken.cs:37]:
```suggestion
        [PreviousVersion("6.2", "BH.Engine.Adapters.CarbonQueryDatabase.Compute.CQDBearerToken(System.String, System.String)")]
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by michaelhoehn** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance
@BHoMBot check required

**Review by FraserGreenroyd — APPROVED** [review]
File changes make sense, great work @michaelhoehn

**Comment by FraserGreenroyd** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 14120375309

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 14120370226

**Pull request merged**